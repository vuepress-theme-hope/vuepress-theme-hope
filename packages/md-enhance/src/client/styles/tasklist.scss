input[type="checkbox"].task-list-item-checkbox {
  position: relative;
  width: 0;
  margin-right: 1.5em;
  cursor: pointer;

  &::after {
    content: " ";

    position: absolute;
    top: 0;

    display: inline-block;

    box-sizing: border-box;

    width: 1em;
    height: 1em;
    padding-left: 0;
    border: 1px solid #ddd;
    border-radius: 2px;

    background-color: #fff;

    text-align: center;

    visibility: visible;

    #{hope-config.$dark-selector} & {
      border-color: #666;
      background-color: #333;
    }
  }

  &:checked {
    &::after {
      content: "";
      border-color: var(--c-brand, #3eaf7c);
      background-color: var(--c-brand, #3eaf7c);

      #{hope-config.$dark-selector} & {
        border-color: var(--c-brand, #3eaf7c);
        background-color: var(--c-brand, #3eaf7c);
      }
    }

    &::before {
      content: "";

      position: absolute;
      top: 0.1em;
      left: 0.4em;
      z-index: 1;

      width: 0.2em;
      height: 0.5em;
      border: solid var(--white);
      border-width: 0 0.15em 0.15em 0;

      transform: rotate(45deg);
    }
  }
}
