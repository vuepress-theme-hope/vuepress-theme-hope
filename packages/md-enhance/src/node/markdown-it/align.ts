import { container } from "./container";

import type { PluginSimple } from "markdown-it";

export const align: PluginSimple = (md) => {
  ["left", "center", "right", "justify"].forEach((name) =>
    md.use((md) =>
      container(md, {
        name,
        openRender: () => `<div style="text-align: ${name}">`,
      })
    )
  );
};
