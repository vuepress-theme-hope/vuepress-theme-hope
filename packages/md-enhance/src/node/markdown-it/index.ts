export * from "./align.js";
export { attrs } from "./attrs.js";
export * from "./chart.js";
export * from "./codeDemo.js";
export * from "./codeTabs.js";
export * from "./container.js";
export * from "./echarts.js";
export * from "./flowchart.js";
export * from "./footnote.js";
export * from "./hint.js";
export * from "./imageMark.js";
export * from "./imageSize.js";
export * from "./katex.js";
export * from "./mathjax.js";
export * from "./lazyLoad.js";
export * from "./mark.js";
export * from "./include.js";
export * from "./mermaid.js";
export * from "./playground/index.js";
export * from "./presentation.js";
export * from "./sub.js";
export * from "./sup.js";
export * from "./tabs.js";
export * from "./tasklist.js";
export * from "./tex.js";
export * from "./vPre.js";
export * from "./stylize.js";
export * from "./uml.js";
export * from "./vuePlayground.js";
