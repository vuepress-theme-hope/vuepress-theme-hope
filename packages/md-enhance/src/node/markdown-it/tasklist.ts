/**
 * Forked from https://github.com/linsir/markdown-it-task-checkbox/blob/master/index.js
 *
 * Copyright (c) 2016, Revin Guillen
 * Modified code Copyright (c) 2016, linsir, MIT License
 *
 * Permission to use, copy, modify, and/or distribute this software for any
 * purpose with or without fee is hereby granted, provided that the above
 * copyright notice and this permission notice appear in all copies.
 *
 * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
 * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
 * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
 * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
 * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
 * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 */

import { default as Token } from "markdown-it/lib/token";
import {
  getParentTokenIndex,
  isInlineToken,
  isParagraphToken,
  isListItemToken,
  setTokenAttr,
} from "./utils";

import type { PluginWithOptions } from "markdown-it";
import type { default as StateCore } from "markdown-it/lib/rules_core/state_core";
import type { TaskListOptions } from "../../shared";

interface TaskListEnv {
  tasklists: number;
}

interface TaskListStateCore extends StateCore {
  env: TaskListEnv;
}
// The leading whitespace in a list item (token.content) is already trimmed off by markdown-it.
// The regex below checks for '[ ] ' or '[x] ' or '[X] ' at the start of the string token.content,
// where the space is either a normal space or a non-breaking space (character 160 = \u00A0).
const startsWithTodoMarkdown = (token: Token): boolean =>
  /^\[[xX \u00A0]\][ \u00A0]/.test(token.content);

const isTaskListItem = (tokens: Token[], index: number): boolean =>
  isInlineToken(tokens[index]) &&
  isParagraphToken(tokens[index - 1]) &&
  isListItemToken(tokens[index - 2]) &&
  startsWithTodoMarkdown(tokens[index]);

const generateCheckbox = (token: Token, id: string, disabled = true): Token => {
  const checkbox = new Token("checkbox_input", "input", 0);

  checkbox.attrs = [
    ["type", "checkbox"],
    ["class", "task-list-item-checkbox"],
    ["id", id],
  ];

  // if token.content starts with '[x] ' or '[X] '
  if (/^\[[xX]\][ \u00A0]/.test(token.content))
    checkbox.attrs.push(["checked", "checked"]);

  if (disabled) checkbox.attrs.push(["disabled", "disabled"]);

  return checkbox;
};

const beginLabel = (id: string): Token => {
  const label = new Token("label_open", "label", 1);

  label.attrs = [
    ["class", "task-list-item-label"],
    ["for", id],
  ];

  return label;
};

const endLabel = (): Token => new Token("label_close", "label", -1);

const todoify = (
  token: Token,
  state: TaskListStateCore,
  options: Required<TaskListOptions>
): void => {
  const id = `task-item-${state.env.tasklists++}`;

  token.children = token.children || [];

  // remove the checkbox syntax letter
  token.children[0].content = token.children[0].content.slice(3);

  if (options.label) {
    // add label
    token.children.unshift(beginLabel(id));
    token.children.push(endLabel());
  }
  // checkbox
  token.children.unshift(generateCheckbox(token, id, options.disabled));
};

export const tasklist: PluginWithOptions<TaskListOptions> = (
  md,
  { disabled = true, label = true } = {}
) => {
  md.core.ruler.after(
    "inline",
    "github-task-lists",
    (state: TaskListStateCore) => {
      const tokens = state.tokens;

      if (!state.env.tasklists) state.env.tasklists = 0;

      for (let i = 2; i < tokens.length; i++) {
        if (isTaskListItem(tokens, i)) {
          todoify(tokens[i], state, { disabled, label });
          setTokenAttr(tokens[i - 2], "class", "task-list-item");
          setTokenAttr(
            tokens[getParentTokenIndex(tokens, i - 2)],
            "class",
            "task-list-container"
          );
        }
      }

      return true;
    }
  );
};
