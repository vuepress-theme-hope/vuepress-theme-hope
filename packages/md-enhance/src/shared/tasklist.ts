export interface TaskListOptions {
  /**
   * 是否禁用 checkbox
   *
   * Whether disable checkbox
   *
   * @default true
   */
  disabled?: boolean;

  /**
   * 是否使用 `<label>` 来包裹文字
   *
   * Whether use `<label>` to wrap text
   *
   * @default true
   */
  label?: boolean;
}
