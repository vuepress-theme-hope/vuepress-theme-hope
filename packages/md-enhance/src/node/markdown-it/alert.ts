import { alert as _alert } from "@mdit/plugin-alert";
import type { ExactLocaleConfig } from "@vuepress/helper";
import { ensureLeadingSlash } from "@vuepress/helper";
import type { PluginWithOptions } from "markdown-it";
import type { MarkdownEnv } from "vuepress/markdown";
import { resolveLocalePath } from "vuepress/shared";

import type { MarkdownAlertLocaleData } from "../typings/index.js";

export type MarkdownItAlertOptions = ExactLocaleConfig<MarkdownAlertLocaleData>;

export const alert: PluginWithOptions<MarkdownItAlertOptions> = (
  md,
  options = {},
) => {
  md.use(_alert, {
    alertNames: ["important", "note", "tip", "warning", "caution", "info"],
    openRender: (tokens, index): string =>
      `<div class="hint-container ${tokens[index].markup}">\n`,
    titleRender: (tokens, index, _options, env: MarkdownEnv) => {
      const type = tokens[index].markup;
      const relativePath = ensureLeadingSlash(env.filePathRelative ?? "");
      const localePath = resolveLocalePath(options, relativePath);

      return `\
<p class="hint-container-title">${
        options[localePath]?.[type as keyof MarkdownAlertLocaleData] || type
      }</p>
`;
    },
    closeRender: () => "</div>\n",
    deep: true,
  });
};
