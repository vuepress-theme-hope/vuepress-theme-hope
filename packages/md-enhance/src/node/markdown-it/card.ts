import { type MarkdownEnv } from "@vuepress/markdown";
import { load } from "js-yaml";
import { type Options, type PluginSimple } from "markdown-it";
import type Token from "markdown-it/lib/token.js";
import {
  entries,
  fromEntries,
  isPlainObject,
  isString,
} from "vuepress-shared/node";

import { stringifyProp } from "./utils.js";
import { logger } from "../utils.js";

export interface CardOptions {
  title: string;
  desc?: string;
  logo?: string;
  link?: string;
  color?: string;
}

const CARD_PROPS = ["title", "desc", "logo", "link", "color"];

const checkCardProps = (config: unknown): CardOptions | null => {
  if (isPlainObject(config) && isString(config["title"]))
    return fromEntries(
      entries(config).filter(
        (item): item is [string, string] =>
          CARD_PROPS.includes(item[0]) && isString(item[1])
      )
    ) as unknown as CardOptions;

  return null;
};

const cardRender = (
  tokens: Token[],
  index: number,
  _options: Options,
  { filePathRelative }: MarkdownEnv
): string => {
  const token = tokens[index];
  const { content, info } = token;

  const language = info.trim().split(":", 2)[1] || "yml";
  let config: unknown = null;

  if (language === "yaml" || language === "yml") config = load(content);
  else if (language === "json")
    try {
      config = <unknown>JSON.parse(content);
    } catch (err) {
      // do nothing
    }

  if (!config) {
    logger.error(
      `Can not parse card config ${language}${
        filePathRelative ? `, found in ${filePathRelative}` : ""
      }.`
    );

    return "";
  }

  const cardData = checkCardProps(config);

  if (cardData) return `<VPCard v-bind='${stringifyProp(cardData)}' />`;

  logger.error(
    `Invalid card config${
      filePathRelative ? ` found in ${filePathRelative}` : ""
    }:
${content}
`
  );

  return "";
};

export const card: PluginSimple = (md) => {
  // Handle ```card  blocks
  const fence = md.renderer.rules.fence;

  md.renderer.rules.fence = (...args): string => {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    const [tokens, index, options, env] = args;
    const { info } = tokens[index];
    const realInfo = info.split(":", 2)[0];

    if (realInfo === "card")
      return cardRender(tokens, index, options, <MarkdownEnv>env);

    return fence!(...args);
  };

  md.renderer.rules["card"] = cardRender;
};
