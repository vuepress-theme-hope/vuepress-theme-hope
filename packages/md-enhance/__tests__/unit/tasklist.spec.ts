import { describe, it, expect } from "vitest";
import MarkdownIt from "markdown-it";
import { tasklist } from "../../src/node/markdown-it/tasklist";

describe("Task list", () => {
  const markdownIt = MarkdownIt({ linkify: true }).use(tasklist);

  it("Should render", () => {
    expect(
      markdownIt.render(`
- [ ] unchecked item 1
- [ ] unchecked item 2
- [ ] unchecked item 3
- [x] checked item 4
`)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
-   [ ] unchecked todo item 1
-   [x] todo item 2
`)
    ).toMatchSnapshot();
  });

  it("Should render nested list", () => {
    expect(
      markdownIt.render(`
1. foo
   * [ ] nested unchecked item 1
   * not a todo item 2
   * not a todo item 3
   * [x] nested checked item 4
2. bar
3. spam
`)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
- foo
  - [ ] nested unchecked item 1
  - [ ] nested unchecked item 2
  - [x] nested checked item 3
  - [X] nested checked item 4
`)
    ).toMatchSnapshot();
  });

  it("Should render ordered list", () => {
    expect(
      markdownIt.render(`
1. [x] checked ordered 1
2. [ ] unchecked ordered 2
3. [x] checked ordered 3
4. [ ] unchecked ordered 4
`)
    ).toMatchSnapshot();
  });

  it("Should not render", () => {
    expect(
      markdownIt.render(`
- [ ]
- [  ] not a todo item 2
- [ x] not a todo item 3
- [x ] not a todo item 4
- [ x ] not a todo item 5
`)
    ).toMatchSnapshot();
  });

  it("Should not render label", () => {
    const markdownItWithOutLabel = MarkdownIt({ linkify: true }).use(tasklist, {
      label: false,
    });

    expect(
      markdownItWithOutLabel.render(`
- [ ] unchecked item 1
- [ ] unchecked item 2
- [ ] unchecked item 3
- [x] checked item 4
`)
    ).toMatchSnapshot();
  });

  it("Should increse id", () => {
    const markdownItWithOutLabel = MarkdownIt({ linkify: true }).use(tasklist, {
      label: false,
    });

    expect(
      markdownItWithOutLabel.render(`
- [ ] unchecked item 1
- [ ] unchecked item 2
- [ ] unchecked item 3
- [x] checked item 4

Some content

- [ ] unchecked item 1
- [ ] unchecked item 2
- [ ] unchecked item 3
- [x] checked item 4
  - [ ] unchecked item 1
  - [ ] unchecked item 2
  - [ ] unchecked item 3
  - [x] checked item 4
`)
    ).toMatchSnapshot();
  });

  it("Should render with items containing other markdown syntax", () => {
    const markdownItWithOutLabel = MarkdownIt({ linkify: true }).use(tasklist, {
      label: false,
    });

    expect(
      markdownItWithOutLabel.render(`
- [ ] unchecked [link](https://example.com)
- [ ] unchecked **item 2**
- [ ] _unchecked_ item 3
- [x] ~~checked item 4~~
`)
    ).toMatchSnapshot();
  });
});
