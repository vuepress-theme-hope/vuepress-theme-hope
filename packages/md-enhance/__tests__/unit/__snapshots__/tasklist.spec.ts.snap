// Vitest Snapshot v1

exports[`Task list > Should increse id 1`] = `
"<ul class=\\"task-list-container\\">
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-0\\" disabled=\\"disabled\\"> unchecked item 1</li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-1\\" disabled=\\"disabled\\"> unchecked item 2</li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-2\\" disabled=\\"disabled\\"> unchecked item 3</li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-3\\" checked=\\"checked\\" disabled=\\"disabled\\"> checked item 4</li>
</ul>
<p>Some content</p>
<ul class=\\"task-list-container\\">
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-4\\" disabled=\\"disabled\\"> unchecked item 1</li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-5\\" disabled=\\"disabled\\"> unchecked item 2</li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-6\\" disabled=\\"disabled\\"> unchecked item 3</li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-7\\" checked=\\"checked\\" disabled=\\"disabled\\"> checked item 4
<ul class=\\"task-list-container\\">
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-8\\" disabled=\\"disabled\\"> unchecked item 1</li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-9\\" disabled=\\"disabled\\"> unchecked item 2</li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-10\\" disabled=\\"disabled\\"> unchecked item 3</li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-11\\" checked=\\"checked\\" disabled=\\"disabled\\"> checked item 4</li>
</ul>
</li>
</ul>
"
`;

exports[`Task list > Should not render 1`] = `
"<ul>
<li>[ ]</li>
<li>[  ] not a todo item 2</li>
<li>[ x] not a todo item 3</li>
<li>[x ] not a todo item 4</li>
<li>[ x ] not a todo item 5</li>
</ul>
"
`;

exports[`Task list > Should not render label 1`] = `
"<ul class=\\"task-list-container\\">
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-0\\" disabled=\\"disabled\\"> unchecked item 1</li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-1\\" disabled=\\"disabled\\"> unchecked item 2</li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-2\\" disabled=\\"disabled\\"> unchecked item 3</li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-3\\" checked=\\"checked\\" disabled=\\"disabled\\"> checked item 4</li>
</ul>
"
`;

exports[`Task list > Should render 1`] = `
"<ul class=\\"task-list-container\\">
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-0\\" disabled=\\"disabled\\"><label class=\\"task-list-item-label\\" for=\\"task-item-0\\"> unchecked item 1</label></li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-1\\" disabled=\\"disabled\\"><label class=\\"task-list-item-label\\" for=\\"task-item-1\\"> unchecked item 2</label></li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-2\\" disabled=\\"disabled\\"><label class=\\"task-list-item-label\\" for=\\"task-item-2\\"> unchecked item 3</label></li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-3\\" checked=\\"checked\\" disabled=\\"disabled\\"><label class=\\"task-list-item-label\\" for=\\"task-item-3\\"> checked item 4</label></li>
</ul>
"
`;

exports[`Task list > Should render 2`] = `
"<ul class=\\"task-list-container\\">
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-0\\" disabled=\\"disabled\\"><label class=\\"task-list-item-label\\" for=\\"task-item-0\\"> unchecked todo item 1</label></li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-1\\" checked=\\"checked\\" disabled=\\"disabled\\"><label class=\\"task-list-item-label\\" for=\\"task-item-1\\"> todo item 2</label></li>
</ul>
"
`;

exports[`Task list > Should render nested list 1`] = `
"<ol>
<li>foo
<ul class=\\"task-list-container\\">
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-0\\" disabled=\\"disabled\\"><label class=\\"task-list-item-label\\" for=\\"task-item-0\\"> nested unchecked item 1</label></li>
<li>not a todo item 2</li>
<li>not a todo item 3</li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-1\\" checked=\\"checked\\" disabled=\\"disabled\\"><label class=\\"task-list-item-label\\" for=\\"task-item-1\\"> nested checked item 4</label></li>
</ul>
</li>
<li>bar</li>
<li>spam</li>
</ol>
"
`;

exports[`Task list > Should render nested list 2`] = `
"<ul>
<li>foo
<ul class=\\"task-list-container\\">
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-0\\" disabled=\\"disabled\\"><label class=\\"task-list-item-label\\" for=\\"task-item-0\\"> nested unchecked item 1</label></li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-1\\" disabled=\\"disabled\\"><label class=\\"task-list-item-label\\" for=\\"task-item-1\\"> nested unchecked item 2</label></li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-2\\" checked=\\"checked\\" disabled=\\"disabled\\"><label class=\\"task-list-item-label\\" for=\\"task-item-2\\"> nested checked item 3</label></li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-3\\" checked=\\"checked\\" disabled=\\"disabled\\"><label class=\\"task-list-item-label\\" for=\\"task-item-3\\"> nested checked item 4</label></li>
</ul>
</li>
</ul>
"
`;

exports[`Task list > Should render ordered list 1`] = `
"<ol class=\\"task-list-container\\">
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-0\\" checked=\\"checked\\" disabled=\\"disabled\\"><label class=\\"task-list-item-label\\" for=\\"task-item-0\\"> checked ordered 1</label></li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-1\\" disabled=\\"disabled\\"><label class=\\"task-list-item-label\\" for=\\"task-item-1\\"> unchecked ordered 2</label></li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-2\\" checked=\\"checked\\" disabled=\\"disabled\\"><label class=\\"task-list-item-label\\" for=\\"task-item-2\\"> checked ordered 3</label></li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-3\\" disabled=\\"disabled\\"><label class=\\"task-list-item-label\\" for=\\"task-item-3\\"> unchecked ordered 4</label></li>
</ol>
"
`;

exports[`Task list > Should render with items containing other markdown syntax 1`] = `
"<ul class=\\"task-list-container\\">
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-0\\" disabled=\\"disabled\\"> unchecked <a href=\\"https://example.com\\">link</a></li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-1\\" disabled=\\"disabled\\"> unchecked <strong>item 2</strong></li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-2\\" disabled=\\"disabled\\"> <em>unchecked</em> item 3</li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" class=\\"task-list-item-checkbox\\" id=\\"task-item-3\\" checked=\\"checked\\" disabled=\\"disabled\\"> <s>checked item 4</s></li>
</ul>
"
`;
